talksapp/main.go

```diff
@@ -43,6 +45,22 @@ func init() {
 	present.PlayEnabled = true
 }
 
+func parseGithubCredentials() string {
+	f, err := os.Open("secret.json")
+	if err != nil {
+		log.Fatalf("open github credentials file secret.json: %v", err)
+	}
+	defer f.Close()
+	var cred struct{ ClientID, ClientSecret string }
+	if err := json.NewDecoder(f).Decode(&cred); err != nil {
+		log.Fatalf("parse github credentials: %v", err)
+	}
+	if cred.ClientID == "" || cred.ClientSecret == "" {
+		log.Fatal("secret.json needs to define ClientID and ClientSecret")
+	}
+	return fmt.Sprintf("client_id=%s&client_secret=%s", cred.ClientID, cred.ClientSecret)
```

Comment:
You should probably use `url.Values` and then its `Encode` method to create this string.